source/unified-test-format/unified-test-format.rst

```diff
@@ -2273,6 +2446,14 @@ match exactly; however, test runners MUST permit variations in document key
 order or otherwise normalize the documents before comparison. If the list of
 documents is empty, the test runner MUST assert that the collection is empty.
 
+The test runner MUST now permit the test's entity map to be accessed and
+the entities to be retrieved. The exact mechanism by which the entities are
+to be retrieved is not prescribed by this specification. As an example,
+the test runner MAY invoke an externally-provided callback with either the
+entity map as is or with only the documents. The test runner MAY also copy
+the document entities from the entity map into a data structure that can
+be later accessed to retrieve the entities.
```

Comment:
Suggested text below losens the language about sharing the entity map itself (since you mention it could be another data structure). I also add a note to consider concurrency and more detail about this all happening at the end of the test before the map is cleared.

> Before clearing the entity map at the end of each test, the test runner MUST allow its entities to be accessed externally. The exact mechanism for facilitating this access is not prescribed by this specification, but drivers should be mindful of concurrency if applicable. As an example, the test runner MAY be configured with a callback method, which will be invoked at the end of each test and provided with the entity map (or an equivalent data structure). As previously discussed in Entity Map (link), test runners MAY restrict access to driver objects and only expose BSON types if necessary.

This assumes that event lists will be BSON types. If you end up creating a special "event list" type (or if you just happen to prefer conciseness), then I'd suggest rewriting the last sentence as:

> As previously discussed in Entity Map (link), test runners MAY restrict access to driver objects if necessary.

Reply:
I used the second version of your suggested language to account for the event list type.